tests/xharness/Xharness.Tests/Logging/Tests/CaptureLogTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.IO;
+using Moq;
+using NUnit.Framework;
+using Xharness.Logging;
+
+namespace Xharness.Tests.Logging.Tests {
+
+	[TestFixture]
+	public class CaptureLogTest {
+
+		string filePath;
+		string capturePath;
+		Mock<ILogs> logs;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			filePath = Path.GetTempFileName ();
+			capturePath = Path.GetTempFileName ();
+			logs = new Mock<ILogs> ();
+			File.Delete (filePath);
+			File.Delete (capturePath);
+		}
+
+		[TearDown]
+		public void TeatDown ()
+		{
+			if (File.Exists (filePath))
+				File.Delete (filePath);
+		}
+
+		
+
+		[Test]
+		public void ConstructorNullFilePath ()
+		{
+			Assert.Throws<ArgumentNullException> (() => {
+				var captureLog = new CaptureLog (logs.Object, null, filePath, false);
+			});
+		}
+
+
+		[Test]
+		public void CaptureRightOrder ()
+		{
+			var ignoredLine = "This lines should not be captured";
+			var logLines = new [] { "first line", "second line", "thrid line" };
+			using (var stream = File.Create (filePath))
+			using (var writer = new StreamWriter (stream)) {
+				writer.WriteLine (ignoredLine);
+			}
+			using (var captureLog = new CaptureLog (logs.Object, capturePath, filePath, false)) {
+				captureLog.StartCapture ();
+				using (var writer = new StreamWriter (filePath)) {
+					foreach (var line in logLines)
+						writer.WriteLine (line);
+				}
+				captureLog.StopCapture ();
+				// get the stream and assert we do have the correct lines
+				using (var captureStream = captureLog.GetReader ()) {
+					string line;
+					while ((line = captureStream.ReadLine ()) != null) {
+						Assert.Contains (line, logLines, line);
+					}
+				}
+			}
+		}
+
+		[Test]
+		public void CaptureMissingFileTest ()
+		{
+			using (var captureLog = new CaptureLog (logs.Object, capturePath, filePath, false)) {
+				Assert.AreEqual (capturePath, captureLog.Path, "capture path");
+				captureLog.StartCapture ();
+				captureLog.StopCapture ();
+			}
+			// read the data that was added to the capture path and  ensure that we do have the name of the missing file
+			using (var reader = new StreamReader (capturePath)) {
+				var line = reader.ReadLine ();
+				StringAssert.Contains (filePath, line, "file path missing");
+			}
+		}
+
+		[Test]
+		public void CaptureWrongOrder ()
+		{
+			Assert.Throws<InvalidOperationException> (() => {
+				using (var captureLog = new CaptureLog (logs.Object, capturePath, filePath, false)) {
+					captureLog.StopCapture ();
+				}
+			});
+		}
+
+		[Test]
+		public void CaptureWrongOrderEntirePath ()
+		{
+			Assert.DoesNotThrow (() => {
+				using (var captureLog = new CaptureLog (logs.Object, capturePath, filePath, true)) {
```

Comment:
I think this should also verify contents of the target file as we are not really testing that functionality anywhere